pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -225,6 +225,10 @@ type SlackConfig struct {
 	// object and accessible by the Prometheus Operator.
 	// +optional
 	APIURL *v1.SecretKeySelector `json:"apiURL,omitempty"`
+	// The file path to read a file which contains the Slack webhook URL.
+	// Ignored if apiURL is set
+	// +optional
+	APIURLFile string `json:"apiUrlFile,omitempty"`
```

Comment:
One issue I see with supporting `APIURLFile` for AlertmanagerConfig resources is that someone can potentially send  notifications to a Stack URL which they didn't provide. The risk is low though because you have to know the file's path in the Alertmanager container. Still it would be a precedent (currently any sensitive material is only referenced via `v1.SecretKeySelector`) and it breaks the kind of isolation we're aiming for with AlertmanagerConfig.

Reply:
Noted, reverted this and moved it so that it can be read from the secret only. Thanks